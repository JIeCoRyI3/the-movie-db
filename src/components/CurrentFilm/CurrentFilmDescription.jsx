import React, { Component } from 'react';
import { withRouter } from 'react-router';
import styles from './CurrentFilmDescription.module.css';

class CurrentFilmDescription extends Component {
    render() {
        console.log(this.props.match.params.id);
        return (
            <div className={styles.currentFilmContainer}>
                <img
                    className={styles.poster}
                    src="https://image.tmdb.org/t/p/w600_and_h900_bestv2/sWgBv7LV2PRoQgkxwlibdGXKz1S.jpg"
                    alt="The Mandalorian"
                    title="The Mandalorian"
                />
                <div className={styles.detailsContainer}>
                    <div className={styles.TitleContainer}>
                        <h4 className={styles.TitleContainerFilmName}>
                            The Mandalorian
                        </h4>
                        <h4>8.5</h4>
                    </div>
                    <div>
                        <h5>2019</h5>
                        <h5>
                            After the fall of the Galactic Empire, lawlessness
                            has spread throughout the galaxy. A lone gunfighter
                            makes his way through the outer reaches, earning his
                            keep as a bounty hunter.
                        </h5>
                    </div>
                </div>
            </div>
        );
    }
}

export default withRouter(CurrentFilmDescription);
