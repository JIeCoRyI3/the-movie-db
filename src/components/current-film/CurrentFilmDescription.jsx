import React from 'react';
import { currentFilmContainer,
        poster,
        detailsContainer,
        detailsContainerTitleContainer,
        detailsContainerTitleContainerTitle } from './CurrentFilmDescription.module.css';

export default function CurrentFilmDescription () {
    return (
        <div className={currentFilmContainer}>
            <img className={poster} src="https://image.tmdb.org/t/p/w600_and_h900_bestv2/sWgBv7LV2PRoQgkxwlibdGXKz1S.jpg" 
                alt="what's wrong"/>
            <div className={detailsContainer}>
                <div className={detailsContainerTitleContainer}>
                    <h4 className={detailsContainerTitleContainerTitle}>The Mandalorian</h4>
                    <h4>8.5</h4>
                </div>
                <div>
                    <h5>2019</h5>
                    <h5>After the fall of the Galactic Empire, lawlessness has spread throughout the galaxy. 
                        A lone gunfighter makes his way through the outer reaches, earning his keep as a bounty hunter.</h5>
                </div>
            </div>
        </div>
    );
};