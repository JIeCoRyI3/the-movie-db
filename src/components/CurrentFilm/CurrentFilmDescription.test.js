import React from 'react';
// import { MemoryRouter, Route } from 'react-router';
// import renderer, { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { CurrentFilmDescription } from './CurrentFilmDescription';

describe('<CurrentFilmDescription/>', () => {
    const match = {
        params: {
            id: 602211,
        },
    };

    const history = {
        push() {
            return 1;
        },
    };

    const response = {
        original_title: 'asd',
        vote_average: 'asd',
        release_date: 'asd',
        overview: 'asd',
        poster_path: 'asd',
    };

    it('should render', () => {
        const component = shallow(
            <CurrentFilmDescription history={history} match={match} />
        );
        expect(component).toMatchSnapshot();
    });

    it('should be', () => {
        let wrapper = shallow(
            <CurrentFilmDescription history={history} match={match} />
        );

        const getMovieMock = jest.fn();
        wrapper.instance().getMovie = getMovieMock.mockReturnValueOnce(
            response
        );
        wrapper.update();
        expect(getMovieMock).toBeCalledWith(602211);
    });
});
