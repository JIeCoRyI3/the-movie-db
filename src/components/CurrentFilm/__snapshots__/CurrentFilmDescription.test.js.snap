// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentFilmDescription/> should render 1`] = `
<div>
  <img
    src="https://image.tmdb.org/t/p/w500/undefined"
  />
  <div>
    <div>
      <h4 />
      <h4 />
    </div>
    <div>
      <h5 />
      <p />
      <div>
        <p>
          Genres:
        </p>
        <p />
      </div>
    </div>
  </div>
</div>
`;
